python/tvm/runtime/module.py

```diff
@@ -30,8 +31,65 @@
 from . import _ffi_api
 
 
-# profile result of time evaluator
-ProfileResult = namedtuple("ProfileResult", ["mean", "results"])
+class BenchmarkResult:
+    """Runtimes from benchmarking"""
+
+    def __init__(self, results: Sequence[float]):
+        """Construct a new BenchmarkResult from a sequence of runtimes.
+
+        Parameters
+        ----------
+        results : Sequence[float]
+            Raw times from benchmarking
+
+        Attributes
+        ----------
+        min : float
+            Minimum runtime in seconds of all results.
+        mean : float
+            Mean runtime in seconds of all results. Note that this mean is not
+            necessarily statistically correct as it is the mean of mean
+            runtimes.
+        median : float
+            Median runtime in seconds of all results. Note that this is not necessarily
+            statistically correct as it is the median of mean runtimes.
+        max : float
+            Maximum runtime in seconds of all results.
+        std : float
+            Standard deviation in seconds of runtimes. Note that this is not necessarily
+            correct as it is the std of mean runtimes.
+        results : Sequence[float]
+            The collected runtimes (in seconds). This may be a series of mean runtimes if
```

Comment:
I think this needs more explanation. Currently a BenchmarkResult object contains no information on the benchmark parameters that were used, and it would be best (IMHO) to avoid there being surprises in terms of the interpretation of the results based on how the object was created. My recommendation would be to either fully document the behavior of what it means to benchmark with 'number > 1' or ensure that the BenchmarkResult object itself contains the benchmark parameters used.